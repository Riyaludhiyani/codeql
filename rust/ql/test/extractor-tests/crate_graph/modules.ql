/**
 * @id module-graph
 * @name Module and Item Graph
 * @kind graph
 */

import rust

query predicate nodes(Item i) {
  i.getParentNode*() = any(Crate m | m.getName() = "test" and m.getVersion() = "0.0.1").getModule()
}

query predicate edges(Item container, Item element) {
  element = container.(Module).getItemList().getAnItem() or
  element = container.(Impl).getAssocItemList().getAnAssocItem() or
  element = container.(Trait).getAssocItemList().getAnAssocItem()
}
