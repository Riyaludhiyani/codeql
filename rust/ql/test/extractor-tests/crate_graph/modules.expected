#-----| fn as_string

#-----| fn length

#-----| fn write

#-----| fn as_string

#-----| fn fmt

#-----| struct X_List

#-----| Static

#-----| mod module
#-----|  -> fn length
#-----|  -> fn write
#-----|  -> struct X_List
#-----|  -> Static
#-----|  -> trait AsString
#-----|  -> Const
#-----|  -> impl AsString for ...::X { ... }
#-----|  -> impl ...::Display for ...::X { ... }
#-----|  -> enum X

#-----| mod crate
#-----|  -> mod module

#-----| trait AsString
#-----|  -> fn as_string

#-----| Const

#-----| impl AsString for ...::X { ... }
#-----|  -> fn as_string

#-----| impl ...::Display for ...::X { ... }
#-----|  -> fn fmt

#-----| enum X
